src/Storefront/Theme/Subscriber/PluginLifecycleSubscriber.php

```diff
@@ -183,9 +183,7 @@ private function doPostActivate(PluginLifecycleEvent $event): void
             $event->getPlugin()->getBaseClass()
         );
 
-        // add plugin configuration to the list of all active plugin configurations
         $configurationCollection = clone $this->storefrontPluginRegistry->getConfigurations();
-        $configurationCollection->add($storefrontPluginConfig);
```

Comment:
this is not always the case as the plugin can define that the container does not need to be rebuild after activation in that case the new plugin config would not already be in the current collection, see https://github.com/shopware/platform/blob/74f3b367e40a8a13a23cf9cc886a77852fb89b51/src/Core/Framework/Plugin/PluginLifecycleService.php#L364

therefore i would suggest that we should check if the plugin is already in the collection and only add it if it is not, better yet would be to adapt the `StorefrontPluginConfigurationCollection` in a way that it ensures that configs are only stored once, e.g. by using the technical name internally as the key for all values

And please adapt your tests to that behaviour, so one testcase that ensures the plugin is not added twice if it is already present in the collection, and one case that it will be added if is not present yet

Reply:
I would suggest checking it explicitly here, optionally also modifying the collection (which i will not do in the next push,  but we can discuss it). This is because the collection here is given through an interface that does not put any constraints on how configurations must be stored, so its easy (through DI) to swap out the registry to an implementation which allows duplicates.